storage/tokudb/ha_tokudb_alter_56.cc

```diff
@@ -503,6 +503,14 @@ enum_alter_inplace_result ha_tokudb::check_if_supported_inplace_alter(
                     tokudb::sysvars::alter_print_error(thd) != 0)) {
                 result = HA_ALTER_INPLACE_EXCLUSIVE_LOCK;
             }
+        } else {
+            // other alter create options could be categorized in two types:
+            // a) options such as KEY_BLOCK_SIZE, BLOCK_FORMAT are not
+            //    supported by TokuDB
+            // b) options such as HA_CREATE_USED_COMMENT, HA_CREATE_USED_DEFAULT_CHARSET
+            //    do not need rebuild engine data
+            // and they all should be ignored
+            result = HA_ALTER_INPLACE_NO_LOCK_AFTER_PREPARE;
```

Comment:
I would prefer that this is more explicit in checking explicitly for the cases where types can be ignored as new types can and will be added in the future, and will automatically be ignored rather than being handled (fully rebuilt) properly.

Reply:
The current logic is try to keep same as InnoDB while handling `Alter_inplace_info::CHANGE_CREATE_OPTION` option, check [innobase_need_rebuild()](https://github.com/percona/percona-server/blob/5.6/storage/innobase/handler/handler0alter.cc#L207-L214).

I agree with you there should be more explicit, prudence is always good in code :)

I'll summit a updated version.

Reply:
Thanks @fungo, we very much appreciate your contribution.

Reply:
@georgelorchpercona happy to contribute and make TokuDB better :)

A updated version has been pushed.

The new patch checks every create option explicitly both in code and test case.